api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.TracerProvider;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.spi.OpenTelemetryFactory;
+import io.opentelemetry.spi.metrics.MeterProviderFactory;
+import io.opentelemetry.spi.trace.TracerProviderFactory;
+import javax.annotation.Nullable;
+
+/**
+ * A global singleton for the entrypoint to telemetry functionality for tracing, metrics and
+ * baggage.
+ *
+ * <p>The global singleton can be retrieved by {@link #get()}. The default for the returned {@link
+ * OpenTelemetry}, if none has been set via {@link #set(OpenTelemetry)}, will be created with any
+ * {@link OpenTelemetryFactory}, {@link TracerProviderFactory} or {@link MeterProviderFactory} found
+ * on the classpath, or otherwise will be default, with no-op behavior.
+ *
+ * <p>If using the OpenTelemetry SDK, you may want to instantiate the {@link OpenTelemetry} to
+ * provide configuration, for example of {@code Resource} or {@code Sampler}. See {@code
+ * OpenTelemetrySdk} and {@code OpenTelemetrySdk.builder} for information on how to construct the
+ * SDK {@link OpenTelemetry}.
+ *
+ * @see TracerProvider
+ * @see MeterProvider
+ * @see ContextPropagators
+ */
+public final class GlobalOpenTelemetry {
```

Comment:
This is a big improvement. I think we need to compare with removing the `getGlobalTracer` type methods in favor of `OpenTelemetry.get().getTracer()` or `OpenTelemetry.getGlobal().getTracer()` or `getGlobalOpentelemetry().getTracer()` (static import). I think they're all fine, latter might lead to more testable user code by reducing the number of static methods involved.

Reply:
I would still not have the "global" functionality on the interface `OpenTelemetry`. Having this separation is a win in my opinion. Not sure I understand what comparison you are asking me to do.

Reply:
@anuraaga any further thoughts at this point? I'm inclined to this this PR the 👍 , but want to make sure your concerns are considered.